pkg/core/resource_graph.go

```diff
@@ -48,8 +48,12 @@ func (rg *ResourceGraph) AddDependency(deployedSecond Resource, deployedFirst Re
 	for _, res := range []Resource{deployedSecond, deployedFirst} {
 		rg.AddResource(res)
 	}
-	rg.underlying.AddEdge(deployedSecond.Id(), deployedFirst.Id())
-	zap.S().Debugf("adding %s -> %s", deployedSecond.Id(), deployedFirst.Id())
+	if cycle, _ := rg.underlying.CreatesCycle(deployedSecond.Id(), deployedFirst.Id()); cycle {
+		zap.S().Errorf("Not Adding Dependency, Cycle would be created from edge %s -> %s", deployedSecond.Id(), deployedFirst.Id())
```

Comment:
I think we should add `(internal error)` or such

Reply:
yeah i think well need to clean this up, ill add a fit and finish task, but for now want this in because its hard to trace these down if theyre from many commits ago

Reply:
I just meant in the error text. If a user sees this, they're going to wonder what they did to create the cycle, and how to fix it; we should let them know it's a bug on our end.